### DISCUSSION TITLE
how to do theming with Tailwind CSS insted of css variables

### DISCUSSION THREAD

**Discussion by milon27** [shadcn-ui/ui#1464] [Q&A]
![image](https://github.com/shadcn-ui/ui/assets/44096479/36733c2e-8318-49a3-ad15-25768510bb74)

when generating a theme from https://ui.shadcn.com/themes here, we only get the values in CSS variables but I am using tailwind CSS classNames How to use these generated themes now?

Thanks.

**Comment by itaim18** [comment]
Basically wherever you declare:
```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```
prefarebly, in a Next.js app in a `globals.css` under app router/pages folder
there you would declare these variables in the @layer base.
These variables would automatically be available as background and text colors and other variables could also be used such as 
```tsx
...
className="bg-popover text-muted border-border"
...
```
Hope this helps :)
